packages/flutter/lib/src/services/autofill.dart

```diff
@@ -0,0 +1,280 @@
+  ///
+  /// * On web, only the first hint is accounted for and will be translated to
+  /// an "autocomplete" string.
+  ///
+  /// See also:
+  ///
+  /// * [AutofillHints], a list of autofill hint strings that is predefined on at
+  /// least one platform.
+  ///
+  /// * [UITextContentType](https://developer.apple.com/documentation/uikit/uitextcontenttype),
+  /// the iOS equivalent.
+  ///
+  /// * Android [autofillHints](https://developer.android.com/reference/android/view/View#setAutofillHints(java.lang.String...)),
+  /// the Android equivalent.
+  ///
+  /// * The [autocomplete](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) attribute,
+  /// the web equivalent.
+  /// {@endtemplate}
+  final List<String> autofillHints;
+
+  /// The current [TextEditingValue] of the [AutofillClient].
+  final TextEditingValue currentEditingValue;
+
+  /// Returns a representation of this object as a JSON object.
+  Map<String, dynamic> toJson() {
+    return <String, dynamic>{
+      'uniqueIdentifier': uniqueIdentifier,
+      'hints': autofillHints,
+      'editingValue': currentEditingValue.toJSON(),
+    };
+  }
+}
+
+/// A client that represents an autofillable input field in the autofill workflow.
+abstract class AutofillClient {
+  /// The unique identifier of this [AutofillClient].
+  ///
+  /// This [AutofillClient] will not participate in autofill until the [autofillId]
+  /// becomes non-null.
+  String get autofillId;
+
+  /// The [TextInputConfiguration] that describes this [AutofillClient].
+  TextInputConfiguration get textInputConfiguration;
+
+  /// Requests this [AutofillClient] update its [TextEditingState] to the given
+  /// state.
+  void updateEditingValue(TextEditingValue newEditingValue);
+}
+
+/// A [TextInputClient] that triggers autofill when focused.
+///
+/// This is typically a [State] or [Element], and should be able to gain focus in
+/// order to trigger autofill.
+abstract class AutofillTrigger extends TextInputClient {
+  /// The [AutofillScope] that this [AutofillTrigger] belongs to.
+  ///
+  /// This [AutofillTrigger] will participate in autofill alone if null.
+  AutofillScope get currentAutofillScope;
+}
+
+mixin AutofillClientMixin implements AutofillClient {
+  @override
+  String get autofillId {
+    final AutofillConfiguration configuration = textInputConfiguration.autofillConfiguration;
+    final String identifier = configuration?.uniqueIdentifier;
+    assert(configuration == null || identifier != null);
+    return identifier;
+  }
+}
+
+/// An ordered group of [AutofillClient]s that are logically connected.
+///
+/// {@template flutter.services.autofill.AutofillScope}
+/// [AutofillClient]s within the same [AutofillScope] are isolated from other
+/// input fields during autofill. That is, when an [AutofillTrigger] gains focus,
+/// only the [AutofillClient]s within the same [AutofillScope] will be visible to
```

Comment:
by [AutofillClient]s you mean the clients within the AutofillTrigger's scope?

Reply:
yes, I think "only the [AutofillClient]s within the same [AutofillScope]" captures that?

Reply:
Yes, looks good.